packages/wallet/src/redux/protocols/indirect-defunding/readme.md

```diff
@@ -0,0 +1,44 @@
+# Indirect De-Funding Protocol
+
+The purpose of this protocol is handle de-funding a channel that has been indirectly funded.
+
+It covers:
+
+- Checking that a channel is closed (either finalized on chain or a conclusion proof exists)
+- Crafting a ledger update that allocates the funds to the players.
+- Waiting for a ledger response from the opponent.
+
+## State machine
+
+### Player A State machine
+
+```mermaid
+graph TD
+  St((start))-->WFAp(WaitForApproval)
+  WFAp-->|Approve|SLU[SendLedgerUpdate]
+  WFAp-->|Rejected|F((failure))
+  SLU-->WFL(WaitForLedgerUpdate)
+  WFL-->|LedgerUpdateReceived|WFAc(WaitForAcknowledgement)
+  WFAc-->|Acknowledged|S((success))
+```
+
+### Player B State machine
+
+```mermaid
+graph TD
+  St((start))-->WFAp(WaitForApproval)
+  WFAp-->|Approve|WFL(WaitForLedgerUpdate)
+  WFAp-->|Rejected|F((failure))
+  WFL-->SLU[SendLedgerUpdate]
+  SLU-->WFAc(WaitForAcknowledgement)
+  WFAc-->|Acknowledged|S((success))
+```
+
+Notes:
+
+- Currently doesn't handle the case where one player approves, one player declines
+
+## Open Questions
+
+1. Is this a top-level protocol? If so what closes the channel?
```

Comment:
Are we expecting the ledger de-funding protocol to be called from a closing protocol of some kind? Right now I don't think we have a protocol for closing the channel. (Previously we had a closing reducer in channel state that handled exchanging the conclude proof).

If so I think the ledger de-funding can be pretty simple. We can assume that a higher level protocol has already 
done the "handshake" so we can assume the ledger de-funding protocol is co-operative and probably don't have to ask for approval.

It seems like we need a higher level protocol to "wrap up" an application. This would be responsible for letting the other wallet know that we want to conclude a game (either by sending a message or a conclude commitment) and triggering the ledger de-funding and then withdrawal protocol. I'm not sure if this is out of scope for this sprint or we thought that's the role the ledger de-funding protocol would play?

Reply:
Yes - I think this will be called from the defunding protocol.